.github/workflows/deploy-playground.yml

```diff
@@ -74,3 +84,18 @@ jobs:
           options: |
             --extra-vars "git_repo=${{ github.event.inputs.gitRepo }} \
                           branch_name=${{ github.event.inputs.branchName }}"
+
+      - name: Save the endpoints file as an artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: endpoints
+          path: devops/aws/endpoints.txt
+
+      - name: Delete CloudFormation Stack if any step failed
+        # Skip if stack already existed
+        if: failure() && steps.stack_exists.outcome != 'failure'
```

Comment:
Would the job progress to this step if "stack_exists" step infact failed ? So it would be sufficient to just check failure.

Reply:
The step `Delete CloudFormation Stack if any step failed` will only run when any of the steps failed except the `stack_exists` step.

Here are the scenarios:
1. Stack does not exist & workflow goes smoothly -> Delete CloudFormations step is skipped
2. Stack exists -> `stack_exists` steps throws failure -> Delete CloudFormation step is skipped
3. Stack does not exist but some ansible step failed -> `failure()` is true  && `steps.stack_exists.outcome` is 'success' -> Delete CloudFormation step runs